src/content/1.7/development/components/form/product/_index.md

```diff
@@ -0,0 +1,239 @@
+---
+title: The Product Form
+menuTitle: Product Form
+weight: 42
+---
+
+# The New Product Form
+{{< minver v="1.7.8" title="true" >}}
+
+In most recent Product page the form is divided into multiple sub forms. Each form has one (or more) dedicated CQRS command to manage its content.
+
+## Creation / Update commands
+
+Except for the initial save, all product modifications are **updates**, which is why `AddProductCommand` is a special case. It requires only:
+
+- the product type
+- localized names (only the name in default language is required)
+
+Once the product is created and its ID has been generated most of the other CQRS commands are **updates** so they will need a `ProductId` identifier.
+They allow **partial updates** to minimize the amount of sent data, of course even partial update require a minimum of consistency and will have validation rules such as required fields in some cases.
+
+## Page form
+
+This form combines different sub form types, each one is managed by a dedicated CQRS command (sometimes more).
+
+| Form type                | Field      | Description                                              | CQRS Command                                           |
+|:-------------------------|:-----------|:---------------------------------------------------------|:-------------------------------------------------------|
+| **BasicInformationType** | `basic`    | Contains the basic product information of the Product | `AddProductCommand` and `UpdateBasicProductCommand` |
+| **RelationshipsType**    | `relationships` | Contains relationships of the product | N/A |
+| **ShortcutType**         | `shortcuts`     | Contains shortcut fields editable in other sub forms but accessible in first tab easily | N/A |
+| **TypeaheadProductPackCollectionType** | `pack_items` | List of products (for Pack of product) | `AddProductToPackCommand` `UpdateProductPackCommand` `RemoveProductFromPachCommand` |
```

Comment:
> RemoveProductFromPachCommand

RemoveProductFrom**Pack**Command
______
Anyway, sorry for bringing this up again, but we need to reconsider whether we create separate add/remove commands or keep it all in one updateProductPack (or something in the middle- like having cleanPackCommand and UpdatePackCommand). Its important to consider which way would be more beneficial for our presentation structure (if we want to refresh the page after updates/deletions or we just handle everything in javascript and pass packed products list for a single command which handles all cases)

Until now we've decided that we should at least create separate command for cleaning the pack.
So the UpdateProductPack will not allow empty array. But if we provide a list without one item, then command will act as RemoveProductFromPack, so the additional command for that might redundant?). Furthermore, we don't need AddProductToPackCommand for same reason.

So the question arises, maybe we should keep only UpdateProductPackCommand (as you've suggested at first I think) which handles all cases depending on packed items array that is passed? (if its empty array we clean the pack, else we insert new pack items instead of old ones)